src/ShortestPaths/ShortestPaths.jl

```diff
@@ -193,12 +197,34 @@ julia> has_negative_weight_cycle(g, d, SPFA())
 false
 ```
 """
-has_negative_weight_cycle(g::AbstractGraph, distmx::AbstractMatrix=weights(g)) = has_negative_weight_cycle(g, distmx, BellmanFord())
-has_negative_weight_cycle(g::AbstractSimpleGraph) = false
+has_negative_weight_cycle(g::AbstractGraph, distmx::AbstractMatrix) = has_negative_weight_cycle(g, distmx, BellmanFord())
+has_negative_weight_cycle(g::AbstractGraph) = false
```

Comment:
Is this really correct? What if we put in a `SimpleWeightedGraph` here?

Reply:
It's not correct at all. You're right.